config/routes.rb

```diff
@@ -165,8 +168,12 @@
     # Letter Opener webmail UI for dev environment
     mount LetterOpenerWeb::Engine, at: 'letter-opener' if Rails.env.development?
 
-    # The Ultimate Catch-All Route! Passes through to page handler,
-    # so that we can have top-level pages - /foo instead of /pages/foo
-    get '*path', to: 'pages#show'
+    # This catch-all route passes through to the Pages controller, allowing
+    # sites to have top-level pages (e.g. /foo instead of /pages/foo).
+    # The constraint here is to avoid also catching the open and click tracking
+    # routes which are appended by the Ahoy::Email engine.
+    get '*path', to: 'pages#show', constraints: lambda { |request|
```

Comment:
Likely a needless time sink, but there's *potentially* (no guarantee!) a workaround to this Ahoy catch all route issue involving these:

- Catch all route workaround with `railties_order` config: https://github.com/ankane/ahoy_email/issues/1
- And/or using the `Ahoy.api = false` setting to disable auto-mounting, so Ahoy::Email can be manually mounted in your routes.rb. https://github.com/ankane/ahoy_email/blob/b478b61cd4c8ee6d8f0bef3894cbfc5d32c5669c/config/routes.rb#L2